lib/components/lenra_button.dart

```diff
@@ -68,7 +83,12 @@ class LenraApplicationButton extends StatelessWidget {
       disabled: disabled ?? false,
       size: size ?? LenraComponentSize.medium,
       type: TypeHelper.fromString(mainStyle),
-      onPressed: () => onButtonPressed(context),
+      onPressed: () {
+        if (submit != null && submit == true) {
```

Comment:
You could add this code to the `onButtonPressed` callback to keep the component props cleaner